examples/engine_headless_ios.rs

```diff
@@ -0,0 +1,117 @@
+//! we need is an engine that _only_ drives the instantiation and
+//! execution of the Wasm module.
+//!
+//! And that, that's a headless engine.
+//!
+//! To achieve such a scenario, a Wasm module must be compiled, then
+//! serialized —for example into a file—, then later, potentially on
+//! another machine, deserialized. The next steps are classical: The
+//! Wasm module is instantiated and executed.
+//!
+//! This example uses a `compiler` because it illustrates the entire
+//! workflow, but keep in mind the compiler isn't required after the
+//! compilation step.
+//!
+//! You can run the example directly by executing in Wasmer root:
+//!
+//! ```shell
+//! cargo run --example engine-headless --release --features "cranelift"
+//! ```
+//!
+//! Ready?
+
+use std::fs::File;
+use std::path::Path;
+use std::str::FromStr;
+use tempfile::NamedTempFile;
+use wasmer::imports;
+use wasmer::wat2wasm;
+use wasmer::Instance;
+use wasmer::Module;
+use wasmer::RuntimeError;
+use wasmer::Store;
+use wasmer::Value;
+use wasmer_compiler::{CpuFeature, Target, Triple};
+use wasmer_compiler_cranelift::Cranelift;
+use wasmer_engine_dylib::Dylib;
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // First step, let's compile the Wasm module and serialize it.
+    // Note: we need a compiler here.
+    let serialized_module_file = {
+        // Let's declare the Wasm module with the text representation.
+        let wasm_bytes = wat2wasm(
+            r#"
+(module
+  (type $sum_t (func (param i32 i32) (result i32)))
+  (func $sum_f (type $sum_t) (param $x i32) (param $y i32) (result i32)
+    local.get $x
+    local.get $y
+    i32.add)
+  (export "sum" (func $sum_f)))
+"#
+            .as_bytes(),
+        )?;
+
+        let compiler_config = Cranelift::default();
+        let triple = Triple::from_str("aarch64-apple-ios")
+            .map_err(|error| RuntimeError::new(error.to_string()))?;
+
+        // Let's define a CPU feature.
+        let mut cpu_feature = CpuFeature::set();
+        cpu_feature.insert(CpuFeature::from_str("sse2")?);
+
+        // Let's build the target.
+        let target = Target::new(triple, cpu_feature);
+        println!("Chosen target: {:?}", target);
+
+        println!("Creating Dylib engine...");
+        let engine = Dylib::new(compiler_config).target(target).engine();
+
+        // Create a store, that holds the engine.
+        let store = Store::new(&engine);
+
+        println!("Compiling module...");
+        // Let's compile the Wasm module.
+        let module = Module::new(&store, wasm_bytes)?;
+        // Here we go. Let's serialize the compiled Wasm module in a
+        // file.
+        println!("Serializing module...");
+        let mut dylib_file = Path::new("./sum.dylib");
+        module.serialize_to_file(dylib_file)?;
+        dylib_file
+    };
+
+    Ok(())
+}
+
+#[test]
+#[cfg(not(any(windows, target_arch = "aarch64", target_env = "musl")))]
```

Comment:
We want it to run on `aarch64`. You must update this please :-).

Reply:
Removed